operator/pkg/apis/infrastructure/v1alpha1/controlplane.go

```diff
@@ -41,47 +42,39 @@ type ControlPlaneList struct {
 // master and etcd are configured to run. By default, KIT uses all the default
 // values and ControlPlaneSpec can be empty.
 type ControlPlaneSpec struct {
-	Infrastructure    `json:",inline"`
 	KubernetesVersion string     `json:"kubernetesVersion,omitempty"`
 	Master            MasterSpec `json:"master,omitempty"`
 	Etcd              ETCDSpec   `json:"etcd,omitempty"`
 }
 
-type Infrastructure struct {
-	VPCCidr        string   `json:"vpccidr,omitempty"`
-	Zones          []string `json:"zones,omitempty"`
-	PrivateSubnets []string `json:"privateSubnets,omitempty"`
-}
-
 // MasterSpec provides a way for the user to configure master instances and
 // custom flags for components running on master nodes like apiserver, KCM and
 // scheduler.
 type MasterSpec struct {
 	Instances         `json:",inline"`
-	Scheduler         *Config `json:"scheduler,omitempty"`
-	ControllerManager *Config `json:"controllerManager,omitempty"`
-	APIServer         *Config `json:"apiServer,omitempty"`
+	Scheduler         *Deployment `json:"scheduler,omitempty"`
```

Comment:
I'd probably model this as the following
```
type MasterSpec struct {
  Scheduler Scheduler
  APIServer APIServer
}

type Scheduler struct {
  corev1.Deployment
  // Additional high level params go here, if any
}

type APIServer struct {
  corev1.Deployment
  // Additional high level params go here, if any
}
```

I'm also curious if it makes more sense to define this as a deployment spec or as a pod spec. I know that we may want to control replicas, so perhaps that's useful. I know ETCD will likely be a statefulset, rather than a deployment. I'm interested to see how this model progresses (i.e. being specific that we are running the Deployment abstraction and not the Pod abstraction).